plugin-api/src/com/mbrlabs/mundus/pluginapi/SceneExtension.java

```diff
@@ -14,13 +14,18 @@
  * limitations under the License.
  */
 
-package com.mbrlabs.mundus.editor.events
+package com.mbrlabs.mundus.pluginapi;
 
-import com.mbrlabs.mundus.commons.scene3d.components.TerrainComponent
+import com.badlogic.gdx.utils.Array;
+import com.mbrlabs.mundus.commons.scene3d.components.TerrainComponent;
+import org.pf4j.ExtensionPoint;
 
-class TerrainVerticesChangedEvent(val terrainComponent: TerrainComponent) {
-    interface TerrainVerticesChangedEventListener {
-        @Subscribe
-        fun onTerrainVerticesChanged(event: TerrainVerticesChangedEvent)
-    }
+public interface SceneExtension extends ExtensionPoint {
```

Comment:
I think a change is needed here. Two options. Should we call this TerrainSceneExtension since it only sends TerrainComponents? Or should the sceneLoaded pass in the entire EditorScene object (which contains all the terrain components)?

Reply:
I've renamed to TerrainSceneExtension.